sklearn/feature_selection/_sequential.py

```diff
@@ -175,18 +181,34 @@ def fit(self, X, y):
             self.n_features_to_select_ if self.direction == 'forward'
             else n_features - self.n_features_to_select_
         )
-        for _ in range(n_iterations):
-            new_feature_idx = self._get_best_new_feature(cloned_estimator, X,
-                                                         y, current_mask)
-            current_mask[new_feature_idx] = True
 
-        if self.direction == 'backward':
-            current_mask = ~current_mask
+        if self.tol is None:
+            for _ in range(n_iterations):
+                new_feature_idx, new_score = self._get_best_new_feature_score(cloned_estimator, X,
+                                                             y, current_mask)
+                current_mask[new_feature_idx] = True
+
+            if self.direction == 'backward':
+                current_mask = ~current_mask
+        else:
+            old_score = 0
```

Comment:
To make sure that the first iteration is always accepted, whatever the value of `self.scoring`.

```suggestion
            old_score = -np.inf
```

Reply:
fix it on https://github.com/scikit-learn/scikit-learn/pull/20145/commits/e47eea7c256df2355e2143decb875ac869076b56